packages/tailwindcss/src/index.ts

```diff
@@ -538,10 +538,8 @@ async function parseCss(
     node.context = {}
   }
 
-  // Replace `@apply` rules with the actual utility classes.
-  features |= substituteAtApply(ast, designSystem)
-
   features |= substituteFunctions(ast, designSystem.resolveThemeValue)
+  features |= substituteAtApply(ast, designSystem)
```

Comment:
Swapped these two around, because `substituteAtApply` will result in a larger tree, which means that the `substituteFunctions` can operate on a smaller tree.

Reply:
yay